frontend/src/app/coworking/ambassador-home/ambassador-home.component.ts

```diff
@@ -39,15 +61,55 @@ export class AmbassadorPageComponent implements OnInit, OnDestroy {
         reservations.filter((r) => r.state === 'CHECKED_IN')
       )
     );
+
+    this.status$ = coworkingService.status$;
   }
 
   ngOnInit(): void {
-    this.refreshSubscription = timer(0, 5000)
+    this.refreshSubscription = timer(0, FIVE_SECONDS)
       .pipe(tap((_) => this.ambassadorService.fetchReservations()))
       .subscribe();
   }
 
   ngOnDestroy(): void {
     this.refreshSubscription.unsubscribe();
   }
+
+  onUsersChanged(users: PublicProfile[]) {
+    if (users.length > 0) {
+      this.coworkingService.pollStatus();
+    }
+  }
+
+  onWalkinSeatSelection(seatSelection: SeatAvailability[]) {
+    if (
+      seatSelection.length > 0 &&
+      this.welcomeDeskReservationSelection.length > 0
+    ) {
+      this.ambassadorService
+        .makeDropinReservation(
+          seatSelection,
+          this.welcomeDeskReservationSelection
+        )
+        .subscribe({
+          next: (reservation) => {
+            this.welcomeDeskReservationSelection = [];
+            // Hack to force the pull of new reservations
+            this.ngOnDestroy();
+            this.ngOnInit();
```

Comment:
Curious, does using `.pollStatus()` here fail?

Reply:
So pollStatus is pulling in the coworking status (seats available), but we're trying to pull in the current reservations immediately (otherwise it's on a 5 second poll). Since there's already a subscription being tracked in ngOnInit/ngOnDestroy, reusing these makes sense. However, this just gave me an idea for cleaning this up a bit. Going ot make a minor refactor and push again.